src/ASBNApp.Frontend/Components/App/Views/MainViewDay.razor

```diff
@@ -0,0 +1,147 @@
+@* Main view to input and edit data, displayed when the user selects the daily view. *@
+
+@using System.Net.Http.Json;
+@inject IASBNDataService dataService;
+@inject ISnackbar Snackbar
+
+<MudPaper Outlined="true" Class="mt-5 pa-5">
+    <MudGrid>
+
+        @* Select date *@
+        <MudItem xs="12">
+            <MudText Typo="Typo.body2">I'm writing a log for...</MudText>
+            <PillSelectorDate @bind-SelectedDate="SelectedDate" />
+        </MudItem>
+
+        @* Select location *@
+        <MudItem xs="6">
+            <MudText Typo="Typo.body2">I was working from...</MudText>
+            <PillSelectorLocation WorkLocationHours="workLocationHours" @bind-SelectedLocation="SelectedLocation" />
+        </MudItem>
+        <MudItem xs="6">
+            <MudNumericField @bind-Value="Hours" Variant="Variant.Text" Label="I worked for..." Min="0" Format="N1" />
+        </MudItem>
+
+        @* Add note and save *@
+        <MudItem xs="12">
+            <MudTextField @bind-Text="Note" T="string" Label="And I learned this today:" AutoGrow MaxLines="5" />
+        </MudItem>
+        <MudItem xs="12">
+            <MudButton Variant="Variant.Filled" Color="Color.Primary" OnClick="SaveData">Save</MudButton>
+        </MudItem>
+
+    </MudGrid>
+</MudPaper>
+
+
+@code {
+    // Variables to bind the selected Date to
+    private DateTime _selectedDate;
+    public DateTime SelectedDate
+    {
+        get => _selectedDate;
+        set
+        {
+            if (value != _selectedDate)
+            {
+                _selectedDate = value;
+                GetDataForDay().GetAwaiter().GetResult();
```

Comment:
ahhh your ui will be not responsive here.

Better use here the `@something` syntax in the razor part of the file. Then you can use in c# a normal async method.

```csharp
<PillSelectorLocation WorkLocationHours="workLocationHours" @onChange="ChangeWeekAsync" @bind-SelectedLocation="SelectedLocation" />

@code{
    public async Task(ChangeEventArgs e){
        // TODO: check if the bind happens before the event
        await GetDataForDay();
    }
}
``

Reply:
and even better give the method a cancellation token. so if the user navigates away while it loads data, it will just cancel the loading.

Reply:
because I see this pattern in the code quite often, try to change it everywhere. It's an antipattern to let properties do tomuch.
You can thing about if it's easier to let the property itself rais an event that others can listen on.

Reply:
their seems to be also a `bind:after` property in razor. Maybe that helps more here:
https://www.mikesdotnetting.com/article/365/executing-async-operations-onchange-in-blazor

Reply:
Trying to get this over the finishing line, captured this in #64 .